lib/charms/vault_k8s/v0/vault_managers.py

```diff
@@ -1447,3 +1448,57 @@ def _get_s3_parameters(self) -> dict[str, str]:
             if isinstance(value, str):
                 s3_parameters[key] = value.strip()
         return s3_parameters
+
+
+class RaftManager:
+    """Encapsulates the business logic for managing the bootstrap of a Vault cluster in Raft mode."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        workload: WorkloadBase,
+        service_name: str,
+        storage_path: str,
+    ):
+        self._juju_facade = JujuFacade(charm)
+        self._workload = workload
+        self._service_name = service_name
+        self._storage_path = storage_path
+
+    def bootstrap(self, node_id: str, address: str) -> None:
+        """Bootstrap a Vault cluster in Raft mode.
+
+        This method will bootstrap a Vault cluster for a single node, by
+        identifying itself as the sole node in the cluster. Additional units
+        may then be added once the cluster is available.
+        """
+        if not self._juju_facade.is_leader:
+            logger.debug("Only leader unit can bootstrap a Vault cluster")
+            raise ManagerError("Only the leader unit can bootstrap a Vault cluster")
+        if not self._juju_facade.planned_units_for_app == 1:
+            raise ManagerError("Bootstrapping a Vault cluster requires exactly one unit")
+
+        self._workload.stop(self._service_name)
+        self._create_peers_json(node_id, address)
+        self._workload.restart(self._service_name)
+
+        logger.info("Vault cluster bootstrapped in Raft mode")
+
+    def _create_peers_json(self, node_id: str, address: str) -> None:
```

Comment:
```suggestion
    def _push_peers_json(self, node_id: str, address: str) -> None:
```
Would be more consistent. (We have `_push_tls_file_to_workload` and `push_autounseal_ca_cert`

Reply:
https://github.com/canonical/vault-k8s-operator/pull/586/commits/8f24774a5f2efc6128de49a742ac1d809006a61a